### PR TITLE
quantile: rename interpolation arg to method

### PR DESCRIPTION
**Pull request by mathause**
numpy/numpy#20327 introduces some changes to `np.quantile` (and related) for the upcoming numpy release (v1.22.0). It renames the `interpolation` keyword to `method` and offers some new interpolation methods. This PR does two things

1. it restores compatibility with numpy 1.22
2. it renames the `interpolation` keyword to `method` in xarray - this change is not strictly necessary but I thought better to be consistent with numpy

- [x] Tests added
- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

(Side note in `dask.array.percentile` the `method` keyword is used differently from the `interpolation` keyword ([docs](https://docs.dask.org/en/stable/generated/dask.array.percentile.html#dask.array.percentile)). However, xarray does not use the dask function.)

---

TODO: need to import `ArrayLike` from npcompat.

### CODE REVIEW

File: xarray/core/dataset.py [hunk 1]
```diff
@@ -6249,17 +6260,13 @@ def quantile(
...
-                            # the former is often more efficient
-                            reduce_dims = None
                         variables[name] = var.quantile(
                             q,
                             dim=reduce_dims,
-                            interpolation=interpolation,
+                            method=method,
                             keep_attrs=keep_attrs,
                             skipna=skipna,
+                            interpolation=interpolation,
```
Comment (mathause) [inline xarray/core/dataset.py:6269]:
```suggestion
```
no necessary

File: xarray/core/dataset.py [hunk 2]
```diff
@@ -6225,15 +6221,30 @@ def quantile(
...
+            warnings.warn(
+                "The `interpolation` argument to quantile was renamed to `method`.",
+                FutureWarning,
+            )
+
+            if method != "linear":
+                raise TypeError("Cannot pass interpolation and method keywords!")
+
+            method = interpolation
+            interpolation = None
```
Comment (mathause) [inline xarray/core/dataset.py:6236]:
```suggestion
```

File: xarray/core/variable.py [hunk 3]
```diff
@@ -1978,8 +1979,14 @@ def no_conflicts(self, other, equiv=duck_array_ops.array_notnull_equiv):
         return self.broadcast_equals(other, equiv=equiv)
 
     def quantile(
-        self, q, dim=None, interpolation="linear", keep_attrs=None, skipna=True
-    ):
+        self,
+        q: np.typing.ArrayLike,
+        dim=None,
```
Comment (mathause) [inline xarray/core/variable.py:1984]: I cannot type check `dim` without changing behaviour for `quantile(q, dim=("x", "y"))`.

File: xarray/core/dataarray.py [hunk 4]
```diff
@@ -3440,18 +3441,13 @@ def quantile(
             Quantile to compute, which must be between 0 and 1 inclusive.
         dim : hashable or sequence of hashable, optional
             Dimension(s) over which to apply quantile.
-        interpolation : {"linear", "lower", "higher", "midpoint", "nearest"}, default: "linear"
```
Comment (dcherian) [inline xarray/core/dataarray.py:3443]: Shall we support `interpolation` for a little longer with a Deprecation Warning?

**Comment by Illviljan** [comment]
`dask.array.percentile` should use the correct numpy name as well now, see https://github.com/dask/dask/pull/8525.

File: xarray/core/dataarray.py [hunk 5]
```diff
@@ -3449,18 +3451,13 @@ def quantile(
...
-
-                - linear: ``i + (j - i) * fraction``, where ``fraction`` is
-                  the fractional part of the index surrounded by ``i`` and
-                  ``j``.
-                - lower: ``i``.
-                - higher: ``j``.
-                - nearest: ``i`` or ``j``, whichever is nearest.
-                - midpoint: ``(i + j) / 2``.
+            use when the desired quantile lies between two data points.
+            See numpy.quantile for available methods.
```
Comment (mathause) [inline xarray/core/dataarray.py:3457]: Is this ok or should we list them explicitly? (The available methods depend on the numpy version which makes this a bit difficult).

**Review by mathause — COMMENTED** [review]
I think this is ready for review.

File: xarray/core/dataarray.py (hunk 5 above)
Comment (jhamman) [reply xarray/core/dataarray.py:3457]: I would lean toward listing the current available methods but if not, we should make sure `numpy.quantile` get's picked up as a link by intersphinx.

**Comment by jhamman** [comment]
+1 on making this change. However, just to note, Pandas seems to still have the the `interpolation` argument (https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.quantile.html) so we will be introducing an inconsistency there.

File: xarray/core/dataarray.py [hunk 6]
```diff
@@ -3426,11 +3427,12 @@ def sortby(
 
     def quantile(
         self,
-        q: Any,
-        dim: Hashable | Sequence[Hashable] | None = None,
-        interpolation: str = "linear",
+        q: ArrayLike,
+        dim: str | Sequence[Hashable] | None = None,
+        method: str = "linear",
```
Comment (Illviljan) [inline xarray/core/dataarray.py:3432]: Can use `Literal["linear", etc]` here instead of `str`.

Comment (mathause) [reply xarray/core/dataarray.py:3432]: I feel that's a bit over the top - can I get away without it?

Comment (max-sixty) [reply xarray/core/dataarray.py:3432]: Agree it would be cool, definitely fine without out too! :)

**Review by dcherian — APPROVED** [review]

**Comment by mathause** [comment]
The rtd failure looks unrelated.

File: xarray/core/dataarray.py [hunk 7]
```diff
@@ -3442,18 +3444,34 @@ def quantile(
...
+                2. "averaged_inverted_cdf" (*)
+                3. "closest_observation" (*)
+                4. "interpolated_inverted_cdf" (*)
+                5. "hazen" (*)
+                6. "weibull" (*)
+                7. "linear"  (default)
+                8. "median_unbiased" (*)
+                9. "normal_unbiased" (*)
+
+            The first three methods are discontiuous.  The following discontinuous
```
Comment (huard) [inline xarray/core/dataarray.py:3462]: Typo in "discontiuous"

**Pull request merged**